C codegen error for `_ := x['nonexisting']`, followed by an if guard

**V doctor:**
```
OS: linux, Ubuntu 20.04.1 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/vmaster
vmodules: /home/delian/.vmodules
vroot: /v/vmaster
vexe: /v/vmaster/v
vexe mtime: 2021-09-15 12:44:41
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 c8d4a64.5bc6cc9
env VFLAGS: "-cc tcc"

Git version: git version 2.25.1
Git vroot status: weekly.2021.37-3-g5bc6cc95-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg vlib/v/checker/tests/require_or_block_sumtype_map.err.vv`
```v
type Abc = string | int

fn main() {
	x := map[string]Abc{}
	_ := x['nonexisting']
	if y := x['nonexisting'] {
		println(y)
	}
}

```

**What did you expect to see?**

A compiled V program with only 1 warning.


**What did you see instead?**
```
vlib/v/checker/tests/require_or_block_sumtype_map.err.vv:5:8: warning: `or {}` block required when indexing a map with sum type value
    3 | fn main() {
    4 |     x := map[string]Abc{}
    5 |     _ := x['nonexisting']
      |           ~~~~~~~~~~~~~~~
    6 |     if y := x['nonexisting'] {
    7 |         println(y)
==================
/tmp/v/require_or_block_sumtype_map.err.7852675258684511247.tmp.c:10439: error: '_t1' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

NB: putting the `	_ := x['nonexisting']` line at the end, produces a working program that can be compiled.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2748,6 +2748,7 @@ fn (mut g Gen) gen_assign_stmt(assign_stmt ast.AssignStmt) {
 				g.expr(val)
 				g.writeln(';}')
 			}
+			g.is_assign_lhs = false
 		} else if assign_stmt.op == .assign
 			&& (is_fixed_array_init || (right_sym.kind == .array_fixed && val is ast.Ident)) {
 			mut v_var := ''

--- a/vlib/v/tests/map_get_assign_blank_test.v
+++ b/vlib/v/tests/map_get_assign_blank_test.v
@@ -1,4 +1,4 @@
-type Abc = string | int
+type Abc = int | string
 
 fn test_map_get_assign_blank() {
 	x := map[string]Abc{}

--- a/vlib/v/tests/map_get_assign_blank_test.v
+++ b/vlib/v/tests/map_get_assign_blank_test.v
@@ -1,0 +1,10 @@
+type Abc = string | int
+
+fn test_map_get_assign_blank() {
+	x := map[string]Abc{}
+	_ := x['nonexisting']
+	if y := x['nonexisting'] {
+		println(y)
+	}
+	assert true
+}

